### ISSUE TITLE
Conditional import based on python version check doesn't work

### ISSUE THREAD

**Issue by Dr-Irv** [microsoft/pyright#4182]
**Describe the bug**
If you have a conditional import based on the python version, pyright still says the import is missing.

This comes up in pandas-stubs where we have some dependent libraries that are not ready for python 3.11 yet, so we want to conditionally import them.

**To Reproduce**
```python
import sys

PY310 = sys.version_info >= (3, 10)
reveal_type(PY310)

if not PY310:
    import foobar
```
pyright reports:
```text
pyvertst.py:7:12 - error: Import "foobar" could not be resolved (reportMissingImports)
pyvertst.py:4:13 - information: Type of "PY310" is "Literal[True]"
```

**Expected behavior**
pyright should not try to do the import based on the version test.

**VS Code extension or command-line**
command line pyright version 1.1.278
Try the above in a python 3.10 environment.

**Comment by erictraut** [comment]
This is expected. You cannot use a variable for the conditional. You need to use the `sys.version_info >= (3, 10)` directly in the `if` statement. 

Pyright (and other type checkers) look for very specific conditional expression forms in the first stage of analysis (the "binder"), which occurs long before the types of any symbols are evaluated.

PEP 484 indicates "type checkers are expected to understand simple version and platform checks" and then gives a few examples. Pyright and all other type checkers support these expression forms.

You'll need to change the above code to the following for it to work in pyright (and other type checkers):

```python
import sys

if not sys.version_info < (3, 10):
    import foobar
```
